lib/sharingmethodinfo.h

```diff
@@ -60,13 +45,13 @@ class TransferMethodInfo
         SupportsMultipleFiles
```

Comment:
Think this now goes to the hints member variable for api compatibility. If we are breaking the api could avoid that separate mechanism.

Then again should we even have the member variables directly to modify, or should there be setter methods + pimplified data instead. Now adding more variables can break the ABI.

Related: the QVariant value(int index) is a strange api too, it's not really index even. Is there actually need for the enum and the related things?

Reply:
> Related: the QVariant value(int index) is a strange api too, it's not really index even. Is there actually need for the enum and the related things?

I think this might be some legacy that we aren't using anymore. I could try dropping it now.

Yeah, getters, setters and keeping data in private class would make sense, I think.